### DISCUSSION TITLE
Adjust anaglyphic (3D stereo) colors in OpenSceneGraph to calibrate to glasses

### DISCUSSION THREAD

**Discussion by madeleinepvincent** [openscenegraph/OpenSceneGraph#1338] [General]
I am rendering a scene in OpenSceneGraph 3.5.1 in anaglyphic model (3D / stereo). The glasses I have are red-cyan. The right eye (cyan) does a great job of masking, and when I look through that eye only, I see just 1 image. The right eye (red) is not as well calibrated, and so the masking is not as complete. I see some of the other image (ghosting).

I could obtain new glasses or possibly adjust the color settings on the monitor, but those don't really solve the problem in the long-term and across many users.

What I want to do is adjust the colors used to generate the anaglyphic image in OpenSceneGraph (or, perhaps, in the underlying OpenGL shader if absolutely necessary, though I'd rather stay up at the OpenSceneGraph level if possible).

So in this case, I'd like to adjust the right-eye red color mask from the default being used.

Thanks in advance,
Madeleine

**Comment by xarray** [comment]
FYI, left/right eye cameras are added as slaves when viewer is set-up. You may get them with getSlave() and set a new ColorMask to their state-sets. You may also consider make use of osg::ColorMatrix to adjust RGB values if you are still using fixed-function pipeline.

**Reply by madeleinepvincent** [reply]
Thanks xarray!  I'll give this a try today and write up how it went.

**Reply by madeleinepvincent** [reply]
Yes, this works as you describe.  Thank you for your answer.
